prometheus/testutil/testutil.go

```diff
@@ -221,6 +221,33 @@ func TransactionalGatherAndCompare(g prometheus.TransactionalGatherer, expected
 	return compareMetricFamilies(got, wanted, metricNames...)
 }
 
+// CollectAndFormat registers the provided Collector with a newly created pedantic Registry. It then calls Gather with
+// that Registry and filters by the provided metricNames. It writes the gathered metrics to a bugger formatted according
+// to the provided format argument
+func CollectAndFormat(c prometheus.Collector, format expfmt.FormatType, metricNames ...string) (*bytes.Buffer, error) {
```

Comment:
Any reason to NOT return bytes or string instead? I would vote for string, it does not matter much. Buffer is weird, usually when you want things to be in buffer, we would need to pass io.Writer in args or something.

Reply:
Nope, no reason I don't think. Although I don't totally understand this:

> we would need to pass io.Writer in args or something

Why is this the case?

Reply:
Perhaps it should be bytes, as the requested format can be protobuf?